Refact node pose.yaml

vitoranello opened the pull request:
Refactor PosePlugin to streamline parameter handling and YAML file reading

vitoranello committed:
Refactor PosePlugin to streamline parameter handling and YAML file reading

vitoranello committed:
Remove comment

Jfsslemos commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -1,10 +1,21 @@
 #!/usr/bin/env python3
+##ADICIONAR YAML DIRETO NO REDIS
```
?

vitoranello replied:
Ty for the review here, xD

Jfsslemos commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -1,10 +1,21 @@
+from ament_index_python.packages import get_package_share_directory
+import yaml
+
+def readYamlFile(file_path: str = None):
+  with open(file_path, 'r') as file:
+    yaml_data = yaml.safe_load(file)['pose']['ros__parameters']['targets']
+  return yaml_data
```
Not needed anymore?

vitoranello replied:
Yes, we need read targes to returned on service. This yaml file would not need to follow the ros guidelines, but by default, I think it is comprehensive to leave it at default.

Jfsslemos replied:
i think it maybe taken out

Jfsslemos commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -21,14 +32,17 @@ def __init__(self, node_name: str = 'pose'):
-    self.readParameters(self.locations)
     self.setStaticPose()
     self.pose_server = self.create_service(GetPose, '/fbot_world/get_pose', self.getPose)
     self.get_logger().info(f"Pose node started!!!")
+    self.get_logger().info(f"Type targets: {type(self.targets)}")
+    self.get_logger().info(f"Pose node locations: {self.targets.keys()}")
```
if youre going to keep the infos add a debug level flag

vitoranello replied:
Its, only for tests on my computer. Just remove.

Jfsslemos reviewed (changes requested):
there are more changes but the correction of the strucuture may solve it

vitoranello committed:
:white_check_mark: Change pose launch file.
:white_check_mark: Change plugin.yaml file to work on any launch.
:white_check_mark: Remove some get_loggers and comments.

kriskappel reviewed (approved):
all good

gadorneles commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -2,9 +2,18 @@
+from ament_index_python.packages import get_package_share_directory
+import yaml
+
+def readYamlFile(file_path: str = None):
```
add ``@brief``

gadorneles commented on fbot_world_bringup/nodes/pose.py:
```diff
@@ -21,11 +30,13 @@ def __init__(self, node_name: str = 'pose'):
     @param node_name The name of the ROS2 node.
```
```suggestion
    @param node_name: The name of the ROS2 node.
```

gadorneles reviewed (approved):
Beautiful!!

Jfsslemos committed:
Update fbot_world_bringup/nodes/pose.py

Co-authored-by: Gabriel Dorneles <dorneles1215@gmail.com>

vitoranello committed:
:sparkles: Refact yaml pose files and fix minor bugs in pose.py

vitoranello committed:
Merge branch 'feat/pose_direct_to_redis' of https://github.com/butia-bots/fbot_world into feat/pose_direct_to_redis

jardeldyonisio reviewed (approved):
Nice refactor, the pose files are so cleaner

gadorneles reviewed (approved):
Looks great!
